File: distance.py
import RPi.GPIO as GPIO
import time

GPIO.setmode(GPIO.BOARD)

TRIG = 12
ECHO = 19

GPIO.setup(TRIG,GPIO.OUT)
GPIO.output(TRIG,0)

GPIO.setup(ECHO,GPIO.IN)

time.sleep(0.1)

print "measuring..."

GPIO.output(TRIG,1)
time.sleep(0.00001)
GPIO.output(TRIG,0)

while GPIO.input(ECHO) == 0:
    pass
start = time.time()

while GPIO.input(ECHO) == 1:
    pass
stop = time.time()

print(stop - start) * 17000

GPIO.cleanup()

File: distance.cc
#include <stdio.h>
#include <time.h>
#include "wiringPi.h"

#define TRIG 12
#define ECHO 19

int main(int argc, char const *argv[]) {
    wiringPiSetupGpio();
    pinMode(TRIG, OUTPUT);
    pinMode(ECHO, INPUT);

    delay(100);
    printf("Measurement in progress...\n");
    clock_t start,end,dur;
    double mseconds = 0;

    digitalWrite(TRIG, HIGH);
    delay(0.00001);
    digitalWrite(TRIG, LOW);
    while (!digitalRead(ECHO)) {
        continue;
    }
    start = clock();

    while (digitalRead(ECHO)) {
        continue;
    }
    end = clock();
    dur = end - start;
    double clocks_per_mill = CLOCKS_PER_SEC / 1000;

    mseconds = dur / clocks_per_mill;
    double distance = mseconds * 0.343;
    printf("Distance: %fm\n", distance);

    return 0;
}
